app/models/concerns/observable.rb

```diff
@@ -42,12 +42,22 @@ def add_observer(observer, func=:update, observers)
       if observers.none?{|k, v| k.is_a?(observer.class)}
         observers[observer] = func
       end
+
       observers
     end
 
     def register_observers
       @@observer_peers ||= {}
-      @@observer_peers = add_observer(Observers::NoticeObserver.new, update_method_name.to_sym, @@observer_peers)
+      @@observer_peers[self.to_s] ||= []
```

Comment:
If you wanted you could initialize as
`@@observer_peers ||= Hash.new([])`

which would give us the default empty array automatically